.ci/setup.sh

```diff
@@ -18,7 +18,7 @@ if [[ $OS_NAME == "macos" ]]; then
         brew install swig
     fi
     brew install graphviz
-    curl -sL -o conda.sh https://repo.anaconda.com/miniconda/Miniconda3-latest-MacOSX-x86_64.sh
+    curl -sL -o conda.sh https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-MacOSX-x86_64.sh
```

Comment:
```suggestion
    curl -sL -o mambaforge.sh https://github.com/conda-forge/miniforge/releases/latest/download/Mambaforge-MacOSX-x86_64.sh
```

Reply:
updated this in https://github.com/microsoft/LightGBM/pull/4953/commits/bc750fbd3523bd6250f35fd4538c0a94c8c03f4b instead of accepting in the browser, to batch all changes together and avoid duplicate CI runs.